src/NHibernate/Impl/AbstractSessionImpl.cs

```diff
@@ -321,15 +321,6 @@ protected bool IsAlreadyDisposed
 
 		protected internal void SetClosed()
 		{
-			try
-			{
-				if (TransactionContext != null)
-					TransactionContext.Dispose();
```

Comment:
No more disposed from session. That is transaction completion responsibility. May otherwise creates race conditions.